### ISSUE TITLE
adb cannot connect on ubuntu-latest today

### ISSUE THREAD

**Issue by finagolfin** [ReactiveCircus/android-emulator-runner#390]
I have been using this action on macos-latest, for hardware acceleration, but now that github switched that to macOS 14 arm64 and I noticed that there is now hw accel on linux too, I switched back to running the emulator on linux.

[That worked fine in my testing yesterday in a pull](https://github.com/finagolfin/swift-android-sdk/actions/runs/8832892043/job/24253081633#step:36:146), but now that I merged the change and had my CI schedule a test run, [adb cannot connect to the emulator anymore](https://github.com/finagolfin/swift-android-sdk/actions/runs/8845858113/job/24297547844#step:35:1021). That has consistently failed a dozen times now, so it's not just some flakiness.

I'm [setting the KVM change](https://github.com/finagolfin/swift-android-sdk/actions/runs/8832892043/job/24253081633#step:35:9) in [both runs](https://github.com/finagolfin/swift-android-sdk/actions/runs/8845858113/job/24297547844#step:34:47), so I don't think that's it, and I don't see any runner image changes since yesterday either.

Anyone else seeing this or know what I'm doing wrong?

**Comment by finagolfin** [comment]
Somebody else took a look at my CI runs and noticed [this error in the failing ones](https://github.com/finagolfin/swift-android-sdk/actions/runs/8845858113/job/24297547844#step:35:113):
```
ERROR   | Not enough space to create userdata partition. Available: 6570.734375 MB at /home/runner/.android/avd/../avd/test.avd, need 7372.800000 MB.
```
Turns out the switch to running the emulator on linux worked initially because I was using some cached SDKs when testing the pull, but once my normal CI runs picked up again, they build the SDKs from scratch, using a lot more disk space and getting the emulator launch to fail for lack of space.

I remedied this by deleting my temporary SDK build directory after a new SDK is built, freeing up GBs of space and this emulator action then works fine again.

**Comment by poponuts** [comment]
hi @finagolfin I know this is closed but can you share the CI steps on **_deleting my temporary SDK build directory after a new SDK is built_**. Also, is this step after `detox build`?

**Comment by finagolfin** [comment]
It could be anything in the CI runner that you don't need at that point, really just depends on what you're doing. Yes, after the build, but just before starting the emulator.

I put in these two lines to [delete large build and source directories](https://github.com/finagolfin/swift-android-sdk/blob/9c214499d1344337b892b6323810ff974551cc34/.github/workflows/sdks.yml#L140) and [a downloaded Swift toolchain that was no longer needed](https://github.com/finagolfin/swift-android-sdk/blob/9c214499d1344337b892b6323810ff974551cc34/.github/workflows/sdks.yml#L316). Another project I'm involved with [removes a bunch of installed Ubuntu packages and system directories when extra build space is needed](https://github.com/termux/termux-packages/blob/a2703e736334432f8f6f18ffc0182b417269492d/.github/workflows/packages.yml#L201), even without running this emulator.

You'll need to remove something like that, whatever you don't need in the CI runner, before running the emulator to avoid this error.
